src/main/java/com/npsproject/it4045npsproject/controller/NPSController.java

```diff
@@ -21,14 +24,25 @@ public NPSController(ParkService parkService, ParkSearchService parkSearchServic
         this.parkSearchService = parkSearchService;
     }
 
-    @GetMapping("/{name}")
+    /**
+     * Return a park based on the name inputted by the end user
+     * @param name string entered by end user in the interface
+     * @return park and park details
+     */
+    @GetMapping("/park/{name}")
     public Park getParkByName(@PathVariable String name) {
         return parkService.getParkDetails(name);
     }
 
-    @PostMapping("/search")
-    public ParkSearchResult searchParks(@RequestBody ParkSearchRequest request) {
-        return parkSearchService.searchParks(request);
+    /**
+     * Return all parks
+     * @return list of all park objects
+     */
+    @GetMapping("/park")
+    @ResponseBody
```

Comment:
Updated to not use ParkSearchService/ParkSearchRequest. A class should not use another class to gather data. Using a new method in the service layer to gather a list of parks from the parks class will improve simplicity and focus.

https://www.baeldung.com/java-clean-code